File: src/main/java/com/abrastat/general/Game.java
package com.abrastat.general;

import com.abrastat.gsc.GSCMoves;
import com.abrastat.gsc.GSCPokemon;
import org.jetbrains.annotations.NotNull;

public interface Game {

    static boolean checkPokemonAreNotFainted(@NotNull GSCPokemon attackingPokemon, GSCPokemon defendingPokemon) {
        return attackingPokemon.getCurrentHP() == 0 || defendingPokemon.getCurrentHP() == 0;
    }

    Moves getLastMoveUsed();
    void setLastMoveUsed(GSCMoves move);

}

File: src/main/java/com/abrastat/general/Game.kt
package com.abrastat.general

interface Game {
    fun getLastMoveUsed(): Move?
    fun setLastMoveUsed(move: Move?)

    companion object {
        @JvmStatic
        fun isPokemonFainted(attackingPokemon: Pokemon, defendingPokemon: Pokemon): Boolean {
            return attackingPokemon.currentHP == 0 || defendingPokemon.currentHP == 0
        }
    }
}